### PR TITLE
Generic sequencing runs

### LINKED ISSUE #117: Add Nanopore support
Add the ability to store oxford nanopore data in IRIDA.  This will include both storage of data, an  uploader, REST API endpoints, and analysis pipelines that use nanopore data as input.

Imported from GitLab issue #551.  Originally posted on 2017/11/08 09:20AM

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
This allows an uploader client to push sequencing runs with different labels to IRIDA instead of them being hardcoded to "MiSeq".  Previously we were working to have dedicated classes to each sequencing run type, but since there's lots of different sequencers and possibly different uses between facilities, at the moment it makes sense to just allow user-defined types.

To test this I suggest getting a dump of a production database and ensuring the upgrade proceeds successfully.  Also creating `SequencingRun`s through the REST API with different `sequencerType` params to verify it works.

## Related issue
Fixes #117 .  Not entirely, there still needs to be analyses for nanopore data, but this will allow the uploaders to send data for other sequencers and it doesn't just appear as "MiSeq" data.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunSequenceFilesController.java [hunk 1]
```diff
@@ -47,7 +43,7 @@ public RESTSequencingRunSequenceFilesController(SequencingRunService service,
 	}
 
 	/**
-	 * Add a relationship between a {@link MiseqRun} and a {@link SequenceFile}.
+	 * Add a relationship between a {@link SequencingRun} and a {@link SequenceFile}.
```
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunSequenceFilesController.java]: Done in 
2f2856e

File: src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java [hunk 2]
```diff
@@ -1,35 +1,25 @@
...
-import ca.corefacility.bioinformatics.irida.web.assembler.resource.ResourceCollection;
-import ca.corefacility.bioinformatics.irida.web.controller.api.RESTGenericController;
-import ca.corefacility.bioinformatics.irida.web.controller.api.projects.RESTProjectsController;
-
-import com.fasterxml.jackson.databind.exc.InvalidDefinitionException;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collection;
 
 /**
  *
```
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/sequencingrun/RESTSequencingRunController.java]: Added that rel back for legacy uploaders in 489e6df

**Comment by JeffreyThiessen** [comment]
The docs around these endpoints need to be updated. `doc/developer/rest/index.md` around line 1040

Other than that everything looks good and is working as intended.

**Comment by tom114** [comment]
Updated the docs and added a deprecation notice to the old endpoint in 77611cd9c9

**Comment by JeffreyThiessen** [comment]
Looks good to go, will merge when travis passes

**Pull request merged**